substratevm/src/com.oracle.svm.agent/src/com/oracle/svm/agent/BreakpointInterceptor.java

```diff
@@ -878,6 +878,76 @@ private static JNIObjectHandle shouldIncludeMethod(JNIEnvironment jni, JNIObject
         return result;
     }
 
+    private static boolean registerCapturingClass(JNIEnvironment jni, Breakpoint bp, InterceptedState state) {
```

Comment:
It is hard for me to follow this code. I think a concise comment about what happens here would be nice. Alternatively, split the method in several calls that have a descriptive name.

Reply:
Comment added as suggested.